Sources/StreamChat/Workers/ChatClientUpdater.swift

```diff
@@ -54,7 +54,11 @@ class ChatClientUpdater {
 
             // Re-create backgroundWorker's since they are related to the previous user.
             client.createBackgroundWorkers()
-
+            
+            // Stop tracking active components
+            client.activeChannelControllers.removeAllObjects()
+            client.activeChannelListControllers.removeAllObjects()
```

Comment:
We should be calling a method on the client, and not know that much from here.

It could also be recreating background workers in an opaque way. This "updater" component knows too much now

Reply:
Yep, we can have a chore task to refactor this 👍